src/Interpreters/MergeTreeTransaction.h

```diff
@@ -0,0 +1,81 @@
+#pragma once
+#include <Interpreters/TransactionVersionMetadata.h>
+#include <boost/noncopyable.hpp>
+#include <Storages/IStorage_fwd.h>
+#include <Storages/TableLockHolder.h>
+#include <Common/Stopwatch.h>
+#include <base/scope_guard.h>
+
+#include <list>
+#include <unordered_set>
+
+namespace DB
+{
+
+class IMergeTreeDataPart;
+using DataPartPtr = std::shared_ptr<const IMergeTreeDataPart>;
+using DataPartsVector = std::vector<DataPartPtr>;
+
+class MergeTreeTransaction : public std::enable_shared_from_this<MergeTreeTransaction>, private boost::noncopyable
```

Comment:
Slightly confusing name because we have `MergeTreeData::Transaction`. Maybe we can add a small comment `// NOTE: `.